### PR TITLE
Add cookie policy

### PR DESCRIPTION
**Pull request by naseberry**
#### What
Implement cookie policy functionality.

#### Ticket
[CACP-247 Implement cookie policy functionality](https://dsdmoj.atlassian.net/browse/CACP-247)

#### Why
Adherence with The Privacy and Electronic Communications Regulations (PECR).

### CODE REVIEW

File: app/webpack/javascripts/cookie-helper.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  // it will return false.
+  return String(getCookie('cookies_preferences_set')) === 'true'
+}
+
+const checkConsentedToCookieExists = () => {
+  return getCookie('cookies_preferences_set') != null
+}
+
+const removeAllPreviousUsedCookies = () => {
+  const constentCookieExist = String(getCookie('cookies_preferences_set')) === 'true'
```
Comment (jsugarman) [inline app/webpack/javascripts/cookie-helper.js]: typo i assume `constent` --> `consent`

Comment (jsugarman) [inline app/webpack/javascripts/cookie-helper.js]: typo i assume `constent` --> `consent`

File: app/webpack/javascripts/cookie-helper.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+  const cookies = document.cookie.split(/;/)
+
+  if (constentCookieExist === false) {
```
Comment (jsugarman) [inline app/webpack/javascripts/cookie-helper.js]: typo i assume `constent` --> `consent`

File: config/locales/en/views/cookies.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+---
+en:
+  cookies:
+    page_title: Cookies on View court data
```
Comment (jsugarman) [inline config/locales/en/views/cookies.yml]: **not a blocker** the app name View court data - should ideally be dynamic

Comment (jsugarman) [reply config/locales/en/views/cookies.yml]: can be take from `service_name`

File: spec/features/help/cookies_spec.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+
+  context 'valid route' do
+    scenario 'cookies page' do
+      visit '/'
+      click_link('Cookies')
+
+      within '.govuk-main-wrapper' do
+        expect(page).to have_css('.govuk-heading-xl', text: I18n.t('cookies.page_title'))
```
Comment (jsugarman) [inline spec/features/help/cookies_spec.rb]: for me i would not test content using `I18n.t('cookies.page_title'))` in a test. its like testing  the implementation. I would write the english text in the test or a regex of it

**Comment by naseberry** [comment]
> LGTM - couple of typos to change before approval but happy to get this in.
> 
> Just some questions:
> 
> 1. Does this adhere to any GDS standard/template? i could not find anything myself!
> 2. have you seen this https://www.gov.uk/help/cookies
> 3. and [this cookie banner](https://components.publishing.service.gov.uk/component-guide/cookie_banner)
> 4. as discussed, whether we [reinventing the wheel](https://api.rubyonrails.org/v5.2.1/classes/ActionDispatch/Cookies.html)

Thanks @jsugarman , in response to you comments:

1. I could not find much aside from the GovUK components guide. Broadly following the ICO. All patterns used are from the GovUK Design system.
2. I have seen this, however as the only cookie we have to consider is GA and as this is an internal service, I have taken a different approach.
3. When this was reviewed, the cookie component did not work, feels like more bloat to the app considering we have installed 'govuk_design_system_formbuilder' and 'govuk-frontend' (although it would be nice to work with components)
4. I am happy to refactor the Cookie implementation at a later time using Rails

File: app/views/layouts/_google_analytics.html.haml [hunk 5]
```diff
@@ -1,7 +1,2 @@
-%script{ async: '', src: "https://www.googletagmanager.com/gtag/js?id=#{ENV['GA_TRACKING_ID']}" }
-
 :javascript
-  window.dataLayer = window.dataLayer || [];
-  function gtag(){dataLayer.push(arguments);}
-  gtag('js', new Date());
-  gtag('config', "#{ENV['GA_TRACKING_ID']}", { 'anonymize_ip': true });
+  let googleTrackingID = "#{ENV['GA_TRACKING_ID']}"
```
Comment (jsugarman) [inline app/views/layouts/_google_analytics.html.haml]: 🎖

**Pull request merged**